sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp

```diff
@@ -1055,8 +1055,14 @@ _detail::WinHttpRequest::~WinHttpRequest()
         "WinHttpRequest::~WinHttpRequest. Closing handle synchronously.");
 
     // Attempt to unregister the status callback. 
-    // if an operation was started the callback will be called regardless, 
-    // as it was initiated before we make the call to unregister
+    // If an operation was started before calling unregister, the callback will be called regardless.
+    // This is a best effort to unregister the callback. 
+    // 
+    // The documentation(https://learn.microsoft.com/windows/win32/api/winhttp/nf-winhttp-winhttpsetstatuscallback)
+    // states : "At the end of asynchronous processing, the application may set the
+    // callback function to NULL. This prevents the client application from receiving additional
+    // notifications." "At the end", not during the process, thus we can still receive notifications
```

Comment:
Two questions: 
1) How do we ensure that we don't receive callbacks after the RegisterWinHttpStatusCallback has returned? I don't see that happening here.
2) How do we get the handle closed callback that the WaitForAction call below is going to expect given that we've unregistered the callback?

Reply:
updated PR per our discussion